rolepermissions/admin.py

```diff
@@ -1,2 +1,31 @@
+from django.conf import settings
+from django.contrib import admin, auth
+from django.contrib.auth.admin import UserAdmin
+from rolepermissions import roles
 
-from django.contrib import admin
+ROLEPERMISSIONS_REGISTER_ADMIN = getattr(settings, 'ROLEPERMISSIONS_REGISTER_ADMIN', False)
+UserModel = auth.get_user_model()
+
+
+class RolePermissionsUserAdminMixin(object):
+    """ Must be mixed in with an UserAdmin class"""
+    def save_related(self, request, form, formsets, change):
+        super(RolePermissionsUserAdminMixin, self).save_related(request, form, formsets, change)
+        # re-load and take a copy of user's newly saved roles
+        user = UserModel.objects.get(pk=form.instance.pk)
+        groups = list(user.groups.all())
+        # reset user's roles to match the list of groups just saved
```

Comment:
This is dangerous because people might be managing permissions regardless of user roles. So this will make those individual changes to be lost. This is rather a "role reset" rather than a simple "sync". Would it be possible to overwrite the adding and removing of `groups` and replace them with `assign` and `remove` role functions?

Reply:
Also, this will need a very descriptive documentation so people understand what is going on under the hood :)

Reply:
I agree, some good docs are required for all 3 features in this PR :-)

The problem this feature is trying to address is:
1) Edit / Create User in Admin;  Assign the User to a Group
2) Group Permissions (defined by Role in code) are not assigned to the User.

I have very limited experience with the use-cases for rolepermissions, but for my use-case, this is important.    I'm not clear what you mean by 

> "Would it be possible to overwrite the adding and removing of groups and replace them with assign and remove role functions?"  

Doesn't the code below use the roles functions?  I'm happy to re-write this in any way that makes sense.

Reply:
Got it. Ok, so despite the ability to manage user permissions by assigning a `role` to a user, it's also possible to manage permissions individually for a user regardless of its `role`. Example:

```
class Nurse(AbstractUserRole):
    available_permissions = {
        'enter_surgery_room`: True,
    }
```
In the common case `Nurses` are permitted to enter the surgery room. But let's say we are instantiating a novice `Nurse` we could do this:
```
new_nurse = User.objects.create_user(...)
assign_role(new_nurse, Nurse)
revoke_permission(new_nurse, 'enter_surgery_room')
```
This way the user still has a `Nurse` role, but is not [yet] allowed in the surgery room.

Now, if you run this script in that situation, the novice nurse will end up with access to the surgery room.

Reply:
Clear -- thanks!
Yes - I can see my code is a like a sledge hammer where something more delicate is required.  I'll give it another shot.

Reply:
> Would it be possible to overwrite the adding and removing of groups and replace them with assign and remove role functions?

This was such a good suggestion @filipeximenes 
Code does exactly this now - basically overrides the adding and removing of groups with logic that assigns and removes roles.  Works very much as expected, even in tricky cases where roles with overlapping permissions are added/removed in the same operation.  

Will push changes to the PR shortly.  Thanks for the excellent code review.